main/php_streams.h

```diff
@@ -125,6 +125,7 @@ typedef struct _php_stream_ops  {
 	int (*cast)(php_stream *stream, int castas, void **ret);
 	int (*stat)(php_stream *stream, php_stream_statbuf *ssb);
 	int (*set_option)(php_stream *stream, int option, int value, void *ptrparam);
+        int (*sync)(php_stream *stream);
```

Comment:
So this addition appears to have implications for some extensions, would appreciate advice from any seasoned contributors in respect of what I'm trying to do here.

Reply:
This is usually handled via the set_option API. See things like PHP_STREAM_OPTION_TRUNCATE_API.

Reply:
@nikic thanks, I've updated the PR with a new implementation based on this advice, looking better?